instrumentation/xxl-job/xxl-job-1.9.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/xxljob/v1_9_2/GlueJobHandlerInstrumentation.java

```diff
@@ -39,29 +38,20 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class ScheduleAdvice {
 
+    @Nullable
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onSchedule(
-        @Advice.FieldValue("jobHandler") IJobHandler handler,
-        @Advice.Local("otelRequest") XxlJobProcessRequest request,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-      Context parentContext = currentContext();
-      request = XxlJobProcessRequest.createGlueJobRequest(handler);
-      context = helper().startSpan(parentContext, request);
-      if (context == null) {
-        return;
-      }
-      scope = context.makeCurrent();
+    public static XxlJobHelper.XxlJobScope onSchedule(
+        @Advice.FieldValue("jobHandler") IJobHandler handler) {
+      return helper().startSpan(XxlJobProcessRequest.createGlueJobRequest(handler));
     }
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
-        @Advice.Return(typing = Assigner.Typing.DYNAMIC) Object result,
-        @Advice.Thrown Throwable throwable,
-        @Advice.Local("otelRequest") XxlJobProcessRequest request,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-      helper().stopSpan(result, request, throwable, scope, context);
+        @Advice.Return(typing = Assigner.Typing.DYNAMIC) @Nullable Object result,
```

Comment:
wondering whether `Assigner.Typing.DYNAMIC` is really needed when assigning to `Object`

Reply:
I wonder the same, I left it as-is for simplicity but it should be safe to remove it in the future.